### PR TITLE
Convert package to CommonJS for broader compatibility

### PR DESCRIPTION
**Pull request by forrest-not-gump**
*Description of changes:*
TLDR; convert core AWSWebsocketProvider class to CommonJS

- convert the core code to be imported and executed as a CommonJS package

- update README

- update test/ example code

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -6,20 +6,43 @@
...
   connect() {
     const region = process.env.AWS_DEFAULT_REGION || 'us-east-1';
-    const credentials = new AWS.EnvironmentCredentials('AWS');
+    let credentials
+    if (this.clientConfig && this.clientConfig.credentials) {
+      if (!validateCredentials(this.clientConfig.credentials)) throw 'Invalid Credentials: Check that your AWS credentials match the standard format';
+      credentials = new AWS.Credentials(this.clientConfig.credentials);
+    } else {
+      credentials = new AWS.EnvironmentCredentials('AWS');
+    }
```
Comment (cayblood) [inline index.js]: Would recommend instead:

```
const creds =
  'clientConfig' in this &&
  'credentials' in this.clientConfig &&
  'accessKeyId' in this.clientConfig.credentials &&
  'secretAccessKey' in this.clientConfig.credentials &&
  this.clientConfig.credentials;
const credentials = (creds && new AWS.Credentials(creds)) || new AWS.EnvironmentCredentials('AWS');
```

This also allows you to remove `validateCredentials()`.

File: index.js [hunk 2]
```diff
@@ -31,7 +54,6 @@ export default class AWSWebsocketProvider extends WebsocketProvider {
     let headers = {
       'Authorization': req.headers['Authorization'],
       'X-Amz-Date': req.headers['X-Amz-Date'],
-      'NodeId': 'nd-ZNOJJ2XNMJAUFP2MSIRL2N45WI',
```
Comment (cayblood) [inline index.js:34]: Was this in my code? Yikes! Thanks for catching that.

File: index.js [hunk 3]
```diff
@@ -40,4 +62,6 @@ export default class AWSWebsocketProvider extends WebsocketProvider {
     this.connection = new Ws(this.url, this.protocol, undefined, headers, this.requestOptions, this.clientConfig);
     this._addSocketListeners();
   }
+
```
Comment (cayblood) [inline index.js]: Would recommend removing blank line

File: README.md [hunk 4]
```diff
@@ -23,15 +22,50 @@ export AWS_SESSION_TOKEN=...
...
 
+You may also provide your credentials directly to the constructor arguments of a new instance of AWSWebsocketProvider():
+```
+const Web3 = require('web3');
+const AWSWebsocketProvider = require('aws-web3-ws-provider');
+
+const credentials = {
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID, 
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY
+}
```
Comment (cayblood) [inline README.md]: add trailing semicolon for consistency

File: README.md [hunk 5]
```diff
@@ -23,15 +22,50 @@ export AWS_SESSION_TOKEN=...
...
+`npm install ethers --save`
+
+```
+const ethers = require('ethers');
+const AWSWebsocketProvider = require('aws-web3-ws-provider');
+
+const credentials = {
+  accessKeyId: process.env.AWS_ACCESS_KEY_ID, 
+  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY
+}
```
Comment (cayblood) [inline README.md]: add trailing semicolon for consistency

**Review by cayblood — CHANGES_REQUESTED** [review]
See inline comments.

File: index.js (hunk 1 above)
Comment (forrest-not-gump) [reply index.js:45]: resolved

File: index.js (hunk 2 above)
Comment (forrest-not-gump) [reply index.js:34]: no worries, seemed to be a remnant from testing

File: README.md (hunk 5 above)
Comment (forrest-not-gump) [reply README.md]: resolved

File: README.md (hunk 4 above)
Comment (forrest-not-gump) [reply README.md]: resolved

File: index.js (hunk 3 above)
Comment (forrest-not-gump) [reply index.js]: resolved

**Comment by forrest-not-gump** [comment]
Resolved requested changes - please review and test locally as needed. Thanks!

**Comment by forrest-not-gump** [comment]
Issues resolved. Please feel free to review and test locally for merge. Thanks!

File: index.js (hunk 1 above)
Comment (forrest-not-gump) [reply index.js]: Resolved - needed to add a check that `clientConfig` is not undefined. When a user does not provide credentials in the `options.clientConfig` object, validation was failing and the `new AWS.EnvironmentCredentials('AWS');` default was not reached due to an error checking for `credentials in undefined`.

**Pull request merged**